### ISSUE TITLE
🐛 Session Replay causes lag on Firefox

### ISSUE THREAD

**Issue by Jonasdoubleyou** [DataDog/browser-sdk#2384]
**Describe the bug**
With Session Replay turned on, when navigating through the page with Firefox, there is a notable lag when navigating through different pages of an SPA (so just the route changes). For some click events, handling the click takes about 1.5s, whereas more than half of the time is spent in `takeSubsequentFullSnapshot` > `addRecord` > `Worker.postMessage` (see the light blue thing in the following Firefox Trace):

![image](https://github.com/DataDog/browser-sdk/assets/23138527/2965bd94-3152-4abf-a14d-2117408d646d)

I suspect this is only observed on Firefox, as [Firefox does the structural clone of the Worker Message on the main thread](https://surma.dev/things/is-postmessage-slow/) (but maybe this also affects other browsers, just noticed so far on Firefox).
Also for what it's worth we have a lot of inline CSS and inline SVGs, maybe this makes cloning the DOM quite large?

Turning off Session Replay with https://github.com/corona-school/user-app/pull/339/files fixed the issue, and no lag is observed afterwards.

Here's the full [Firefox Profiler Trace](https://profiler.firefox.com/docs/#/) with Session Replay:
[Firefox 2023-08-19 21.06 profile.json.gz](https://github.com/DataDog/browser-sdk/files/12386661/Firefox.2023-08-19.21.06.profile.json.gz)

And here without for comparison:
[Firefox 2023-08-19 21.47 profile.json.gz](https://github.com/DataDog/browser-sdk/files/12386662/Firefox.2023-08-19.21.47.profile.json.gz)

**To Reproduce**
Feel free to contact me as jonas.wilms under the domain lern-fair.de to access our staging environment for reproduction (we can also turn on Session Replay there for reproduction). Unfortunately as I can only speculate on the reasons I have no minimal reproducible example. 

**Expected behavior**
Session Replay adds no overhead that users can notice.

Thanks!
Jonas - Lern-Fair

**Comment by Ross-Blakeney** [comment]
Facing the same issue here. We are having to remove session replay from our apps because of this.

Firefox slows down and encounters `InternalError: too much recursion`.
There did seem to be noticeable slow down in chromium based browsers as well. Once we removed session replay all lag disappeared. This is anecdotal at the moment until we profile more browsers.

**Comment by amortemousque** [comment]
Hello @Jonasdoubleyou,
Thanks for the detailed feedback. To give us access to your staging environment could you reach out to [Datadog support](https://help.datadoghq.com/hc/en-us) so we keep track of everything.

**Comment by Jonasdoubleyou** [comment]
Hi @amortemousque, I managed to get it working in the meantime.
We extensively use inline SVGs as well as some iframes, I used the class "dd-privacy-hidden" to exclude them (not sure how large the effect was of this). Additionally we use the Zoom SDK in our application, which appears to use some Inline CSS Loader, so once the SDK is imported, a few inline `<style>` elements are added to the header which have a few megabytes each. I added some lazy loading so that the styles are only added when actually joining a Zoom Meeting. The requests to `/replay` I see now have only a few hundred bytes in size, so this seems to have fixed it really well.

Feel free to close this issue, however I'd appreciate if things like the "dd-privacy-hidden" class and performance considerations in general could be documented somewhere.

Best Regards,
Jonas
